### ISSUE TITLE
Warnings with TeX Live 2018

### ISSUE THREAD

**Issue by nad** [borisveytsman/acmart#294]
I get warnings when using acmart with TeX Live 2018:
```
$ cat bug.tex 
\documentclass[acmsmall]{acmart}
\begin{document}
Text
\end{document}
$ pdflatex bug.tex | egrep 'acmart|warning'
(/usr/local/stow/texlive/texmf-dist/tex/latex/acmart/acmart.cls
Document Class: acmart 2018/04/14 v1.53 Typesetting articles for the Associatio
pdfTeX warning: pdflatex: ToUnicode: value out of range [0,10FFFF]: 740074
pdfTeX warning: pdflatex: ToUnicode: value out of range [0,10FFFF]: 660074
pdfTeX warning: pdflatex: ToUnicode: value out of range [0,10FFFF]: 540068
$ pdflatex --version | head -1
pdfTeX 3.14159265-2.6-1.40.19 (TeX Live 2018)
```
The warnings seem to be related to the commit that closed #167.

**Comment by nad** [comment]
Another set of warnings:
```
$ cat bug.tex 
\documentclass[acmsmall]{acmart}
\begin{document}
\title{T}
\maketitle
\end{document}
$ pdflatex bug.tex | grep -A1 'hyperref Warning'
Package hyperref Warning: Option `pdflang' has already been used,
(hyperref)                setting the option has no effect on input line 4.
--
Package hyperref Warning: Option `pdfdisplaydoctitle' has already been used,
(hyperref)                setting the option has no effect on input line 4.
```

**Comment by Blaisorblade** [comment]
Tldr: the `ToUnicode` warnings re #167 seem fixed by adding a space to separate the output characters, and that seems the correct thing to write, but I couldn't reproduce #167 to confirm it's still fixed:

```diff
-\pdfglyphtounicode{t_t}{00740074}
-\pdfglyphtounicode{f_t}{00660074}
-\pdfglyphtounicode{T_h}{00540068}
+\pdfglyphtounicode{t_t}{0074 0074}
+\pdfglyphtounicode{f_t}{0066 0074}
+\pdfglyphtounicode{T_h}{0054 0068}
```

## In detail

These warnings seem to disappear by adding a space since they're meant to represent *two* characters — I tried this by cargo-culting the text in https://tex.stackexchange.com/q/65821/1340 (if that were wrong, people would have pointed it out), but it's also used in `glyphtounicode.tex` itself (for me, that's in `/usr/local/texlive/2018/texmf-dist/tex/generic/pdftex/glyphtounicode.tex`, but grep `glyphtounicode` in your logs to find it).

But I can't test it properly because I failed to reproduce #167 in the limited time I had — `written` typesets and copy-pastes fine in `acmsmall` and `sigplan` formats (I've also checked author names, they seem fine too). Dunno what's the involved font needed to trigger this.

**Comment by arjenpdevries** [comment]
Hi @Blaisorblade I get the second error as well (pdflang option when using \maketitle).

**Comment by Blaisorblade** [comment]
@kberry Thanks for the info and suggestions. Since acmart doesn't require TeXLive 2019 (and likely shouldn't), IMHO it should keep the `pdfglyphtounicode` calls even if they're also upstreamed to `libertine.sty` or to `glyphtounicode.tex` (not sure). Just opened #298 (with credit) for the quickfix. But I fear I can't take care of the upstreaming!

@arjenpdevries see #296 and #297.

**Comment by ozross** [comment]
Hi Karl, Boris, Paolo

On Jul 16, 2018, at 4:26 AM, kberry <notifications@github.com<mailto:notifications@github.com>> wrote:


pdftex from TL'18 does require spaces to separate Unicode code points in the second arg of \pdfglyphtounicode. This was always documented but never enforced until the heuristic splitting at every four chars was shown to be deficient.

Therefore, Boris, yes, you should change acmart.dtx. That is what is making the now-invalid \pdfglyphtounicode calls, therefore that is what should be fixed.


Whether the upstream glyphtounicode.tex should be changed, I don't know. I have the impression that these glyph names are peculiar to libertine (like the common ligatures f_f, f_f_i, etc.), but then again, maybe there would be no harm in defining them.

I just had a look in  glyphtounicode.tex  to see which glyph names use multiple unicode characters.
Here are some of them:

SCI-3:links ross$ grep " " `kpsewhich glyphtounicode.tex` | grep pdfglyph - | grep "{f"
\pdfglyphtounicode{ff}{0066 0066}
\pdfglyphtounicode{ffi}{0066 0066 0069}
\pdfglyphtounicode{ffl}{0066 0066 006C}
\pdfglyphtounicode{fi}{0066 0069}
\pdfglyphtounicode{finalkafqamats}{05DA 05B8}
\pdfglyphtounicode{finalkafqamatshebrew}{05DA 05B8}
\pdfglyphtounicode{finalkafsheva}{05DA 05B0}
\pdfglyphtounicode{finalkafshevahebrew}{05DA 05B0}
\pdfglyphtounicode{fl}{0066 006C}

Since naming like  f_f  is documented as part of the Adobe Glyph List, it would make good sense
to add variants:  (but see further discussion below)

\pdfglyphtounicode{f_f}{0066 0066}
\pdfglyphtounicode{f_f_i}{0066 0066 0069}
\pdfglyphtounicode{f_f_l}{0066 0066 006C}
\pdfglyphtounicode{f_i}{0066 0069}
\pdfglyphtounicode{f_l}{0066 006C}

also

\pdfglyphtounicode{L_L}{004C 004C}
\pdfglyphtounicode{S_S}{0053 0053}
\pdfglyphtounicode{s_s}{0073 0073}%   same as  SSsmall
\pdfglyphtounicode{c_t}{0063 0074}
\pdfglyphtounicode{l_l}{006C 006C}
\pdfglyphtounicode{s_t}{0073 0074}


Just for comparison, here are the relevant lines from  acmart.cls  (before adjustment)

SCI-3:links ross$ grep pdfglyph `kpsewhich acmart.cls`
\pdfglyphtounicode{f_f}{FB00}
\pdfglyphtounicode{f_f_i}{FB03}
\pdfglyphtounicode{f_f_l}{FB04}
\pdfglyphtounicode{f_i}{FB01}
\pdfglyphtounicode{t_t}{00740074}
\pdfglyphtounicode{f_t}{00660074}
\pdfglyphtounicode{T_h}{00540068}

Don’t know why  f_l  is not included also, for  U+FB02
and   s_t  for  U+FB06

Not sure whether   longs_t   for  U+FB05  is ever needed?
Can’t hurt, I suppose.


This does beg the question of whether  glyphtounicode.tex  should
change the values for  ff, etc.  to reference  U+FB**  values.



—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://protect-au.mimecast.com/s/pCuJCGv0Z6f1VZNoC7pSdc?domain=github.com>, or mute the thread<https://protect-au.mimecast.com/s/3vAnCJyBZ6tq7LWPUzoxhc?domain=github.com>.

Hope this helps.

   Ross

**Comment by Blaisorblade** [comment]
@ozross @kberry we merged the change here, anyone of you up for upstreaming it? Neither https://ctan.org/pkg/pdftex nor https://ctan.org/pkg/libertine have a proper bug tracker (the latter doesn't even list contacts for Bob Tennent), and I'm not the expert here.

I've filed an issue (possibly not relevant) for the libertinus fork (https://github.com/libertinus-fonts/libertinus/issues/199), but I can't help further.
